compiler/rustc_middle/src/ty/list.rs

```diff
@@ -198,6 +198,10 @@ impl<'a, T: Copy> IntoIterator for &'a List<T> {
 
 unsafe impl<T: Sync> Sync for List<T> {}
```

Comment:
You could remove this impl now.

Reply:
`collect_crate_mono_items` needs `Vec<MomoItems>` (which uses List inside) to impl `rayon::IntoParallelIterator`. I think we can keep it here?

Reply:
Ah. It should be fine to keep.